lighthouse-core/audits/legacy-javascript.js

```diff
@@ -0,0 +1,364 @@
+      'Array.prototype.findIndex',
+      'Array.prototype.forEach',
+      'Array.prototype.includes',
+      'Array.prototype.indexOf',
+      'Array.prototype.lastIndexOf',
+      'Array.prototype.map',
+      'Array.prototype.reduce',
+      'Array.prototype.reduceRight',
+      'Array.prototype.some',
+      'ArrayBuffer',
+      'DataView',
+      'Date.now',
+      'Date.prototype.toISOString',
+      'Date.prototype.toJSON',
+      'Date.prototype.toString',
+      'Float32Array',
+      'Float64Array',
+      'Function.prototype.bind',
+      'Function.prototype.name',
+      'Int16Array',
+      'Int32Array',
+      'Int8Array',
+      'Map',
+      'Number.isInteger',
+      'Number.isSafeInteger',
+      'Number.parseFloat',
+      'Number.parseInt',
+      'Object.assign',
+      'Object.create',
+      'Object.defineProperties',
+      'Object.defineProperty',
+      'Object.entries',
+      'Object.freeze',
+      'Object.getOwnPropertyDescriptor',
+      'Object.getOwnPropertyDescriptors',
+      'Object.getOwnPropertyNames',
+      'Object.getPrototypeOf',
+      'Object.isExtensible',
+      'Object.isFrozen',
+      'Object.isSealed',
+      'Object.keys',
+      'Object.preventExtensions',
+      'Object.seal',
+      'Object.setPrototypeOf',
+      'Object.values',
+      'Promise',
+      'Reflect.apply',
+      'Reflect.construct',
+      'Reflect.defineProperty',
+      'Reflect.deleteProperty',
+      'Reflect.get',
+      'Reflect.getOwnPropertyDescriptor',
+      'Reflect.getPrototypeOf',
+      'Reflect.has',
+      'Reflect.isExtensible',
+      'Reflect.ownKeys',
+      'Reflect.preventExtensions',
+      'Reflect.set',
+      'Reflect.setPrototypeOf',
+      'Set',
+      'String.fromCodePoint',
+      'String.prototype.codePointAt',
+      'String.prototype.endsWith',
+      'String.prototype.includes',
+      'String.prototype.padEnd',
+      'String.prototype.padStart',
+      'String.prototype.repeat',
+      'String.prototype.startsWith',
+      'String.prototype.trim',
+      'String.raw',
+      'Uint16Array',
+      'Uint32Array',
+      'Uint8Array',
+      'Uint8ClampedArray',
+      'WeakMap',
+      'WeakSet',
+    ].map(polyfillName => {
+      const parts = polyfillName.split('.');
+      const object = parts.length > 1 ? parts.slice(0, parts.length - 1).join('.') : null;
+      const property = parts[parts.length - 1];
+      return {
+        name: polyfillName,
+        expression: this.buildPolyfillExpression(object, property),
+      };
+    });
+  }
+
+  /**
+   * @return {Pattern[]}
+   */
+  static getTransformPatterns() {
+    return [
+      {
+        name: '@babel/plugin-transform-classes',
+        expression: 'Cannot call a class as a function',
+      },
+      {
+        name: '@babel/plugin-transform-regenerator',
+        expression: /regeneratorRuntime\.a?wrap/.source,
+      },
+      {
+        name: '@babel/plugin-transform-spread',
+        expression: /\.apply\(void 0/.source,
```

Comment:
this is too broad for us to use :/ is there any other reliable pattern that's more specific?

Reply:
ah, good catch. I'll take a look.

Reply:
done! `/\.apply\(void 0,\s?_toConsumableArray/`